pkg/skaffold/schema/latest/config.go

```diff
@@ -234,10 +248,12 @@ type HelmConventionConfig struct {
 // Artifact represents items that need to be built, along with the context in which
 // they should be built.
 type Artifact struct {
-	ImageName    string            `yaml:"image,omitempty"`
-	Workspace    string            `yaml:"context,omitempty"`
-	Sync         map[string]string `yaml:"sync,omitempty"`
-	ArtifactType `yaml:",inline"`
+	ImageName            string                `yaml:"image,omitempty"`
+	Workspace            string                `yaml:"context,omitempty"`
+	Sync                 map[string]string     `yaml:"sync,omitempty"`
+	BuilderPlugin        *BuilderPlugin        `yaml:"plugin,omitempty"`
+	ExecutionEnvironment *ExecutionEnvironment `yaml:"executionEnvironment,omitempty"`
```

Comment:
now that we have `Init()` can't we leave this out and move it back later _if_ there is a need for different artifacts to run in different environments? I feel like we introduced it because it _might_ be useful, but there was no real need for it except the technical need to send somehow the information of the ExecEnv to the plugin.